### DISCUSSION TITLE
How to integrate nextjs `basePath` with next-intl?

### DISCUSSION THREAD

**Discussion by hemengke1997** [amannn/next-intl#242] [Q&A]
I'm using next-intl with `app dir`.
I tried to set basePath, but the `next-intl/middleware` always set locale prefix on url which leading to 404.

Is there any workround to solve this? Thanks everyone.

**Comment by amannn** [comment]
Hey, thanks for starting this discussion! This is definitely a valid feature request, I'll convert this to an issue. 

I haven't explored this topic in detail yet, but possibly, we'd have to wait for the lead from Next.js here:

> `basePath` has been removed. The alternative will not be part of useRouter. It has not yet been implemented.

(from [the upgrade guide](https://beta.nextjs.org/docs/upgrade-guide#step-5-migrating-routing-hooks))

I just saw that in the `pages` folder with i18n routing, Next.js appends the locale to the base path (e.g. `domain.tld/base-path/en/about` and even `domain.tld/base/path/en/about`). Is that also what you need?

**Comment by amannn** [comment]
Closed in favour of https://github.com/amannn/next-intl/issues/243
